src/HotChocolate/Core/src/Types/Utilities/DefaultTypeConverter.cs

```diff
@@ -144,7 +144,7 @@ private bool TryConvertInternal(Type from, Type to, object source, out object? c
             Type target,
             [NotNullWhen(true)] out ChangeType? converter)
         {
-            if (source == target || target == typeof(object))
+            if (target == typeof(object) || target.IsAssignableFrom(source))
```

Comment:
@michaelstaib this fixes the issue at hand and the tests show no issues. Seems like it could have side-effects though. What are your thoughts on this change?

Reply:
Actually thought about this as well. IsAssignableFrom() includes ==, but also returns true if target inherits from source. From my experience and tests the performance is not an issue, but I see your point as it could potentially be something to cause issues further down the line.

Reply:
I will write some performance tests for this one.